src/status_im2/contexts/chat/messages/link_preview/events.cljs

```diff
@@ -41,32 +42,93 @@
    {}))
 
-(rf/defn handle-community-resolved
-  {:events [:chat.ui/community-resolved]}
-  [{:keys [db] :as cofx} community-id community]
-  (rf/merge cofx
-            (cond-> {:db (update db :communities/resolve-community-info dissoc community-id)}
+(defn community-resolved
+  [{:keys [db]} [community-id community]]
+  (when community
+    {:db (update db :communities/resolve-community-info dissoc community-id)
+     :fx [[:dispatch [:communities/handle-community community]]
+          [:dispatch
+           [:chat.ui/cache-link-preview-data (community-link community-id) community]]]}))
 
-              (some? community)
-              (assoc :dispatch
-                     [:chat.ui/cache-link-preview-data (community-link community-id) community]))
-            (models.communities/handle-community community)))
+(rf/reg-event-fx :chat.ui/community-resolved community-resolved)
 
-(rf/defn handle-community-failed-to-resolve
-  {:events [:chat.ui/community-failed-to-resolve]}
-  [{:keys [db]} community-id]
+(defn community-failed-to-resolve
+  [{:keys [db]} [community-id]]
   {:db (update db :communities/resolve-community-info dissoc community-id)})
 
-(rf/defn resolve-community-info
-  {:events [:chat.ui/resolve-community-info]}
-  [{:keys [db]} community-id]
-  {:db            (assoc-in db [:communities/resolve-community-info community-id] true)
-   :json-rpc/call [{:method     "wakuext_requestCommunityInfoFromMailserver"
-                    :params     [community-id]
-                    :on-success #(rf/dispatch [:chat.ui/community-resolved community-id %])
-                    :on-error   #(do
-                                   (rf/dispatch [:chat.ui/community-failed-to-resolve community-id])
-                                   (log/error "Failed to request community info from mailserver"))}]})
+(rf/reg-event-fx :chat.ui/community-failed-to-resolve community-failed-to-resolve)
+
+(defn fetch-community
+  [{:keys [db]} [community-id]]
+  (when community-id
+    {:db            (assoc-in db [:communities/resolve-community-info community-id] true)
+     :json-rpc/call [{:method     "wakuext_fetchCommunity"
+                      :params     [{:CommunityKey    community-id
+                                    :TryDatabase     true
+                                    :WaitForResponse true}]
+                      :on-success (fn [community]
+                                    (rf/dispatch [:chat.ui/community-resolved community-id community]))
+                      :on-error   (fn [err]
+                                    (rf/dispatch [:chat.ui/community-failed-to-resolve community-id])
+                                    (log/error {:message
+                                                "Failed to request community info from mailserver"
+                                                :error err}))}]}))
+
+(schema/=> fetch-community
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args
+     [:schema [:? :string]]]]
```

Comment:
It would be nice if we could name the arguments, something like what `:catn` gives, so that the args (which is usually what we care the most) is more readable than `[:? :string]`. Just food for our thought, I don't have a specific suggestion yet.